src/backend/executor/cypher_create.c

```diff
@@ -594,6 +596,20 @@ static Datum create_vertex(cypher_create_custom_scan_state *css,
                      errmsg("agtype must resolve to a vertex")));
         }
 
+        // extract the label agtype field
+        label_value = GET_AGTYPE_VALUE_OBJECT_VALUE(v, "label");
+
+        // extract the label name and label id
+        label_name = label_value->val.string.val;
+        if (strlen(label_name) == 0)
```

Comment:
Can this function `get_label_id_from_entity` replace this IF block?

Reply:
I believe we would have to pass the `graph_name` in both `create_vertex` and `merge_vertex` to make it work. However, this might slow down the execution as we would then call `search_graph_name_cache(graph_name)` (scan) to retrieve the `graph_oid`, even though we already have it in the context of `create_vertex` and `merge_vertex`

Reply:
I am not sure if we need the IF block if there is already a function that does the job.

Reply:
I believe we don't have the graph_name in the context of `merge_vertex` and create_vertex`, and this function needs the `graph_name` to retrive the `graph_oid` that we already have

Reply:
You may create a function similar to `get_label_id_from_entity` that takes graph_oid as an input.